### PR TITLE
Update variable specificity to include process scoping

### PR DESCRIPTION
**Pull request by shaunmarx**
This PR adds the associated documentation when scoping variables for runbooks.

### CODE REVIEW

**Comment by shaunmarx** [comment]
Thanks for reviewing @wordlee - you wouldn't have found any mentions of this since it's a new addition to accommodate scoping variables for runbooks.

**Comment by wordlee** [comment]
👍 Sorry, I should have been clearer :) I couldn't find any other places where we list the precedence order :)

File: docs/deployment-process/operations-runbooks/index.md [hunk 1]
```diff
@@ -37,11 +37,8 @@ You can create many runbooks per project and share the [project variables](/docs
 
 ### Current Limitations
 
-**Prompted Variables**
-Prompted variables are not currently supported for the Runbooks feature, but we aim to have support for prompted variables in the near future. There is no workaround at this time.
-
 **Scoping**
-You cannot scope project variables to either a Deployment or Runbook Process currently, but we do aim to support this in the near future. A workaround is to namespace your Runbook-specific project variables. For example: Your project variable could be named `Runbook.Foo`. This naming prefix may help you easily see and consume "Runbook" variables in your Runbook Process.
+- You cannot scope project variables to a Deployment Process Action as well as a Runbook Process action currently, but we do aim to support this in the near future.
```
Comment (MarkSiedle) [inline docs/deployment-process/operations-runbooks/index.md]: Could we update the title of this to be "Scoping to Steps/Actions" so it's more clear this is all about actions. I think most customers think about steps, not actions, so we should use the word Steps here I think

File: docs/deployment-process/operations-runbooks/index.md [hunk 2]
```diff
@@ -87,8 +84,7 @@ Record the HTTP requests made by the Octopus UI to see how we build your runbook
 
 For this early access feature, there are a number of things we decided to not include initially (so we could get the feature in your hands sooner). Some known limitations that are not currently supported are:
 
-- Prompted variables (if your project includes prompted variables, they will be ignored for runbooks).
-- Variable scoping to runbooks (there's no way to scope variables between your project's deployment process vs runbook processes).
+- There is no way to scope to a runbook action.
```
Comment (MarkSiedle) [inline docs/deployment-process/operations-runbooks/index.md]: Again, I know technically it's an action scope, but for customers, they think in terms of steps. So either step/action or just step would be more user-friendly reading.

File: docs/deployment-process/variables/index.md [hunk 3]
```diff
@@ -106,6 +107,7 @@ Since variables can be scoped in many different ways, there needs to be a predic
 1. The target tenant-tag (if tenant-features are enabled).
 1. The target environment.
 1. The target channel (if channels are enabled).
+1. The current deployment process or runbook.
```
Comment (MarkSiedle) [inline docs/deployment-process/variables/index.md]: The current deployment process or runbook process.

File: docs/deployment-process/variables/index.md [hunk 4]
```diff
@@ -115,6 +117,19 @@ Variable scoping also works like CSS rules; a value scoped twice is more specifi
 
 If two variables are scoped equally, Octopus will choose project-defined variables ahead of library-defined ones. If this still does not resolve the conflict the result is non-deterministic and you should not depend on a specific value being used. Instead, you should take care when scoping variables so that they are unlikely to conflict.
 
+### Mutually Exclusive Scopes {#Mutuallyexclusive-scopes}
```
Comment (MarkSiedle) [inline docs/deployment-process/variables/index.md]: This is awesome 👍

File: docs/deployment-process/variables/index.md [hunk 5]
```diff
@@ -115,6 +117,19 @@ Variable scoping also works like CSS rules; a value scoped twice is more specifi
...
+
+Scopes of the same type can generally be thought of as a grouping of logical OR expressions while scopes of different types evaluate as AND expressions between the groups of scopes. As an example, a variable scoped to the `Development` environment, `Production` environment as well as the `Default` channel would evaluate as `(Development OR Production) AND (Default)`. This example results in a specific scope which requires a channel in order for the variable to evaluate. This means that the variable will not be usable in the context of a runbook as channels do not apply in this context.
+
+There are two distinct scoping scenarios which need to be explicitly called out, one of which we encountered above, these are:
+
+* Scoping to both a channel as well as a runbook
```
Comment (MarkSiedle) [inline docs/deployment-process/variables/index.md]: Maybe always refer to this as "runbook process"

File: docs/deployment-process/operations-runbooks/index.md (hunk 1 above)
Comment (shaunmarx) [reply docs/deployment-process/operations-runbooks/index.md]: Did we want to change all occurrences to use step? i.e.
- You cannot scope project variables to a Deployment Process *step* as well as a Runbook Process *step* currently, but we do aim to support this in the near future.

**Pull request merged**